test/libsolidity/semanticTests/functionTypes/external_functions_with_calldata_args_assigned_to_function_pointers_with_memory_type.sol

```diff
@@ -0,0 +1,10 @@
+contract C {
+    function g(string calldata) external {}
+
+    function main() view external returns (function (string memory) external) {
+        function (string memory) external ptr = this.g;
+        return ptr;
+    }
+}
+// ----
+// main() -> -28758981899283947592798738437886595599040542797332825194040240852133054775296
```

Comment:
I have not idea if this result is correct or not and the point of a semantic test is to make sure the code calculates correct results. It would be better to add some return values and have `g` call the function it's getting and return its result (could be just the string it's getting as its argument). We could then check that result.

Reply:
Made relevant changes in the latest push.